python/servo/mutation/test.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2013 The Servo Project Developers. See the COPYRIGHT
+# file at the top-level directory of this distribution.
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+import fileinput
+import re
+import subprocess
+import sys
+import os
+import random
+DEVNULL = open(os.devnull, 'wb')
+
+
+def mutate_random_line(file_name):
+    line_numbers = []
+    for line in fileinput.input(file_name):
+        if re.search(r'\s&&\s', line):
+            line_numbers.append(fileinput.lineno())
+    if len(line_numbers) == 0:
+        return -1
+    else:
+        mutation_line_number = line_numbers[random.randint(0, len(line_numbers) - 1)]
+        for line in fileinput.input(file_name, inplace=True):
+            if fileinput.lineno() == mutation_line_number:
+                line = re.sub(r'\s&&\s', ' || ', line)
+            print line.rstrip()
+        return mutation_line_number
+
+
+def mutation_test(file_name, tests):
+    local_changes_present = subprocess.call('git diff --quiet {0}'.format(file_name), shell=True)
+    if local_changes_present == 1:
+        print "{0} has local changes, please commit/remove changes before running the test".format(file_name)
+    else:
+        mutated_line = mutate_random_line(file_name)
```

Comment:
As mentioned above, I would prefer to see the `mutate_random_line` call passing in the mutation strategy here: `mutate_random_line(file_name, r'\s&&\s', ' || ')`. That'll save a bit of refactoring if/when we get to a point where we want this function to take multiple mutations in some way -- imagine `mutation_test(file_name, tests, mutations)` for instance.